packages/compass-e2e-tests/tests/database-collections-tab.test.ts

```diff
@@ -28,9 +25,38 @@ describe('Database collections tab', function () {
   });
 
   it('contains a list of collections', async function () {
-    expect(await browser.existsEventually(Selectors.CollectionsGrid)).to.eq(
-      true
+    const collectionsGrid = await browser.$(Selectors.CollectionsGrid);
+    await collectionsGrid.waitForDisplayed();
+
+    const collectionSelectors = ['json-array', 'json-file', 'numbers'].map(
+      (collectionName) => Selectors.collectionCard('test', collectionName)
+    );
+
+    for (const collectionSelector of collectionSelectors) {
+      const collectionElement = await browser.$(collectionSelector);
+      await collectionElement.waitForExist();
+    }
```

Comment:
Does it make a difference whether this is for-loop or Promise.all() (do `.waitForExist()` calls resolve immediate if the element already exists)?

Reply:
Pretty much immediately unless you want to count the next tick or whatever happens when awaiting a promise.

Reply:
I'm basically using waitForExist as an implicit expect. You can't immediately check if they already exist because the code will execute before the page updated. So in reality there will be a short wait before the first one as the page transitions and renders. There are lots of ways to write this but this feels simple and compact to me. Also accomplishes both the waiting and the assertions in one go.